gnucash/gnome/gnc-plugin-page-account-tree.c

```diff
@@ -1521,14 +1692,22 @@ gnc_plugin_page_account_tree_cmd_delete_account (GtkAction *action, GncPluginPag
                 ta = gnc_account_sel_get_account(GNC_ACCOUNT_SEL(trans_mas));
                 ta_match = gnc_account_get_currency_or_parent (ta) == account_currency;
             }
-            if (sa_mas && gtk_widget_is_sensitive(sa_mas))
-                saa = gnc_account_sel_get_account(GNC_ACCOUNT_SEL(sa_mas));
             if (sa_trans_mas && gtk_widget_is_sensitive(sa_trans_mas))
+            {
                 sta = gnc_account_sel_get_account(GNC_ACCOUNT_SEL(sa_trans_mas));
-            if (!ta_match)
+                sta_match = gnc_account_get_currency_or_parent (sta) == account_currency;
+            }
+            if (sa_mas && gtk_widget_is_sensitive(sa_mas))
+            {
+                saa = gnc_account_sel_get_account(GNC_ACCOUNT_SEL(sa_mas));
+                saa_match = gnc_account_get_currency_or_parent (saa) == account_currency;
+            }
+            if (!ta_match || !sta_match || !saa_match)
             {
-                // The currencies in the account that we're deleting and the account we're moving transactions to don't match.
-                char* message = g_strdup_printf (_("Account %s does not have the same currency as the account you're deleting.\nAre you sure you want to do this?"), gnc_account_get_full_name (ta));
+                char* message = NULL;
+                char* a_name = (!ta_match) ?  gnc_account_get_full_name (ta) : (!sta_match) ?  gnc_account_get_full_name (sta) :
+                                              gnc_account_get_full_name (saa);
+                message = g_strdup_printf (_("Account %s does not have the same currency as the account you're deleting.\nAre you sure you want to do this?"), a_name);
                 GtkWidget *error_dialog = gtk_message_dialog_new (GTK_WINDOW(window),
                                                                   GTK_DIALOG_DESTROY_WITH_PARENT,
                                                                   GTK_MESSAGE_ERROR,
```

Comment:
This is the travis failure, similar to the one in the other PR, move GtkWidget *error_dialog; to the beginning of the if branch and then change this line to error_dialog = ... and do not forget to realign the following lines.